src/charm.py

```diff
@@ -111,25 +108,19 @@ def _reconcile(self) -> None:
             logger.exception("Error initializing the lightkube client.")
             raise RuntimeError("Error initializing the lightkube client.") from exc
 
-        try:
-            gateway_resource_definition = GatewayResourceDefinition.from_charm(self)
-        except InvalidCharmConfigError:
-            logger.exception("Invalid charm config.")
-            self.unit.status = BlockedStatus("Invalid charm configuration")
-            return
-
-        tls_certificates_relation = self._tls.get_tls_relation()
-        if not tls_certificates_relation:
-            self.unit.status = BlockedStatus("Waiting for TLS.")
-            return
+        config = CharmConfig.from_charm(self)
+        gateway_resource_definition = GatewayResourceDefinition.from_charm(self)
+        # This line is currently here to validate the existence of the certificates relation.
+        # This charm state component will be used by the upcoming SecretResourceManager.
+        _ = TLSInformation.from_charm(self)
```

Comment:
Nitpick: Would `_` be required here? (feel free to resolve!)

Reply:
I removed it to avoid any confusion